public/app/features/dashboard-scene/scene/layout-default/DefaultGridLayoutManager.tsx

```diff
@@ -412,6 +413,20 @@ export class DefaultGridLayoutManager
     });
   }
 
+  public static trackIfEmpty(grid: SceneLayout) {
```

Comment:
I think we should be a bit more explicit about `trackIfEmpty` implementation. I suggest adding this as `DashboardLayoutManager` interface method so that each layout must implement it. Even if you want to re-use tracking function between `DefaultGridLayoutManager` and `ResponsiveGridLayoutManager` as you do it in this PR, i'd rather have that function as a utility, rather than `ResponsiveGridLayoutManager` referencing `DefaultGridLayoutManager.trackIfEmpty` function - it's quite confusing and makes the layouts depend on each other.

Other thing to think about is whether or not it has to be a behavior. I would consider creating a base layout class that adds tracking explicitly through activation handler, rather than forcing us to remember to add this behavior any tima a layout is created.

Reply:
That's a good point on the base class + activation handler, didn't give it much thought but would indeed be better than a behavior! I'll refactor it to use that.

Reply:
let me see if we there is a simpler option

Reply:
@torkelo @dprokop how about something like this? removed the behavior and moved to an activation handler in an abstract class

Reply:
@dprokop @mdvictor what do you think of https://github.com/grafana/grafana/pull/100189 ?

Reply:
got so caught up in using what was, like the isEmpty flag, didn't even think about any other options. 

This looks great, much nicer! I'm closing this one in favor of the one you opened.